Incorrect ETL_ASSERT_FAIL argument types

**Describe the bug**
`ETL_ASSERT_FAIL` is called with a `raw string literal (const char*)` in `libs/bsw/middleware` but the macro expects its argument e to be of type `const etl::exception&`.
This causes type check errors when migrating `libs/bsw/middleware` to Bazel.

Affected call sites:
- [libs/bsw/middleware/src/middleware/core/ProxyBase.cpp:151](https://github.com/eclipse-openbsw/openbsw/blob/main/libs/bsw/middleware/src/middleware/core/ProxyBase.cpp#L151)
- https://github.com/eclipse-openbsw/openbsw/blob/main/libs/bsw/middleware/src/middleware/core/SkeletonBase.cpp#L181
- https://github.com/eclipse-openbsw/openbsw/blob/main/libs/bsw/middleware/test/src/time/mock/SystemTimerProviderMock.cpp#L34

```bash
In file included from libs/3rdparty/etl/include/etl/algorithm.h:42,
                 from libs/3rdparty/etl/include/etl/array.h:35,
                 from bazel-out/k8-fastbuild/bin/libs/bsw/middleware/_virtual_includes/middleware/middleware/core/Message.h:7,
                 from bazel-out/k8-fastbuild/bin/libs/bsw/middleware/_virtual_includes/middleware/middleware/core/IClusterConnection.h:6,
                 from bazel-out/k8-fastbuild/bin/libs/bsw/middleware/_virtual_includes/middleware/middleware/core/SkeletonBase.h:5,
                 from libs/bsw/middleware/src/middleware/core/SkeletonBase.cpp:3:
libs/bsw/middleware/src/middleware/core/SkeletonBase.cpp: In member function 'void middleware::core::SkeletonBase::checkCrossThreadError(uint32_t) const':
libs/bsw/middleware/src/middleware/core/SkeletonBase.cpp:181:29: error: invalid initialization of reference of type 'const etl::exception&' from expression of type 'const char [42]'
  181 |             ETL_ASSERT_FAIL("Skeleton cross thread violation detected.");
      |   
```

I believe these errors are ignored on the Cmake-side due to how ETL is included in the build graph (via -isystem). Bazel is stricter and produces a type check error.

**Steps to reproduce the bug**
A Bazel POC branch can be provided later

**Expected behavior and actual behavior**
`ETL_ASSERT_FAIL` is called with an `etl::exception object`, for example via the `ETL_ERROR_GENERIC` helper macro, so that the code compiles and behaves correctly under all ETL preprocessor configurations.

**Environment**
Operating System: Docker container development
Build System/Tools: Docker container default

--- a/libs/bsw/middleware/src/middleware/core/ProxyBase.cpp
+++ b/libs/bsw/middleware/src/middleware/core/ProxyBase.cpp
@@ -148,7 +148,7 @@ void ProxyBase::checkCrossThreadError(uint32_t const initId) const
                 initId,
                 currentTaskId);
 
-            ETL_ASSERT_FAIL("Proxy cross thread violation detected.");
+            ETL_ASSERT_FAIL(ETL_ERROR_GENERIC("Proxy cross thread violation detected."));
         }
     }
 }

--- a/libs/bsw/middleware/src/middleware/core/SkeletonBase.cpp
+++ b/libs/bsw/middleware/src/middleware/core/SkeletonBase.cpp
@@ -178,7 +178,7 @@ void SkeletonBase::checkCrossThreadError(uint32_t const initId) const
                 initId,
                 currentTaskId);
 
-            ETL_ASSERT_FAIL("Skeleton cross thread violation detected.");
+            ETL_ASSERT_FAIL(ETL_ERROR_GENERIC("Skeleton cross thread violation detected."));
         }
     }
 }

--- a/libs/bsw/middleware/test/src/time/mock/SystemTimerProviderMock.cpp
+++ b/libs/bsw/middleware/test/src/time/mock/SystemTimerProviderMock.cpp
@@ -31,7 +31,7 @@ uint32_t getCurrentTimeInMs()
         return gSystemTimerProviderMockPtr->getCurrentTimeInMs();
     }
 
-    ETL_ASSERT_FAIL("SystemTimerProviderMock is not set.");
+    ETL_ASSERT_FAIL(ETL_ERROR_GENERIC("SystemTimerProviderMock is not set."));
 }
 
 uint32_t getCurrentTimeInUs()
@@ -41,7 +41,7 @@ uint32_t getCurrentTimeInUs()
         return gSystemTimerProviderMockPtr->getCurrentTimeInUs();
     }
 
-    ETL_ASSERT_FAIL("SystemTimerProviderMock is not set.");
+    ETL_ASSERT_FAIL(ETL_ERROR_GENERIC("SystemTimerProviderMock is not set."));
 }
 
 } // namespace middleware::time

